src/permutations.jl

```diff
@@ -9,14 +9,52 @@ export
     permutations
 
 
-struct Permutations{T}
-    a::T
-    t::Int
+struct PermutationIterator{T}
+    data::T
+    length::Int
+end
+
+function has_repeats(state::Vector{Int})
```

Comment:
here and below since `state` is `Vector{Int}` you can assume 1-based indexing, but what you did here is OK so up to you how to treat it.

Reply:
oh, I see!